lib/ansible/modules/cloud/azure/azure_rm_batchaccount.py

```diff
@@ -0,0 +1,331 @@
+#!/usr/bin/python
+#
+# Copyright (C) 2019 Junyi Yi (@JunyiYi)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_batchaccount
+version_added: "2.9"
+short_description: Manages a Batch Account on Azure.
+description:
+    - Create, update and delete instance of Azure Batch Account.
+
+options:
+    resource_group:
+        description:
+        - The name of the resource group in which to create the Batch Account.
+        required: true
+    name:
+        description:
+        - The name of the Batch Account.
+        required: true
+    location:
+        description:
+        - Specifies the supported Azure location where the resource exists.
+    auto_storage_account:
+        description:
+        - The ID of the Batch Account auto storage account.
+    key_vault_reference:
+        description:
+        - A reference to the Azure key vault associated with the Batch account.
+        suboptions:
+            id:
+                description:
```

Comment:
this one should probably also support the usual Ansible object ref patterns via `normalize_resource_id`- not sure if the URL is always obtainable from the keyvault object, but maybe we should just look it up off the actual keyvault instead of requiring it to be specified? Seems strange that the underlying REST API needs it, but I'm not sure exactly how it's used.

Reply:
I just checked Azure CLI and they simply have:

keyvault: The KeyVault name or resource ID to be used for an account with a pool allocation mode of 'User Subscription'.

so we can safely assume that keyvault id should be sufficient.

I think keyvault URL may be just an alternative way of specifying a keyvault.